File: src/CalcViewModel/ViewState.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

#include "pch.h"
#include "ViewState.h"

namespace CalculatorApp
{
    namespace ViewState
    {
        Plat::StringReference Snap(L"Snap");
        Plat::StringReference DockedView(L"DockedView");

        bool IsValidViewState(Plat::String ^ viewState)
        {
            return viewState->Equals(ViewState::Snap) || viewState->Equals(ViewState::DockedView);
        }
    }
}

File: src/Calculator.Shared/ViewModels/ViewState.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CalculatorApp
{
    public class ViewState
    {
        public const string Snap = "Snap";
        public const string DockedView = "DockedView";

        public bool IsValidViewState(string viewState) => viewState.Equals(ViewState.Snap) || viewState.Equals(ViewState.DockedView);
    }
}
